src/io/hexdump.cr

```diff
@@ -0,0 +1,59 @@
+# IO object that prints an hexadecimal dump of all transfered data.
+#
+# Especially useful for debugging binary protocols on an IO, to understand
+# better when and how data is sent or received.
+#
+# By default `IO::Hexdump` won't print anything; you must specify which of
+# `read`, `write` or both you want to print.
+#
+# Example:
+# ```
+# require "io/hexdump"
+# io = IO::Hexdump.new(socket, output: STDERR, read: true)
+# ```
+#
+# When data is read from `io` it will print something akin to the following on
+# STDERR:
+# ```
+# 00000000  50 52 49 20 2a 20 48 54  54 50 2f 32 2e 30 0d 0a  PRI * HTTP/2.0..
+# 00000010  0d 0a 53 4d 0d 0a 0d 0a                           ..SM....
+# 00000000  00 00 00 04                                       ....
+# 00000000  00                                                .
+# 00000000  00 00 00 00                                       ....
+# ```
+module IO
+  class Hexdump
+    include IO
+
+    def initialize(@io : IO, @output : IO = STDERR, @read = false, @write = false)
```

Comment:
I think it's OK to use STDERR. For example today I was coding something and did `ruby app.rb > data.json` and wanted to debug it at the same time. For debugging I used `STDERR.puts ...` so I would get the debug data in the terminal but I would still get the output data in `data.json` without mixing it with debug code.